CenterForOpenScience/osf.io
# [feature] Update visibility of registration schemas [PLAT-1246]

#### Purpose
Prevent certain registration schemas from being returned by the API (and on the Registries discovery page) by either removing them from the database or marking them as not visible. See the ticket for details of which are being deleted and which are being hidden (as well as which have registrations attached to them on production).

#### Changes
* Adds a `visible` field to `RegistrationSchema`
  * `active`: whether or not the schema is accepting submissions
  * `visible`: whether or not the schema should be surfaced in the API (and everywhere else)

#### QA Notes
* Confirm that none of the following schemas are ever returned by the API (especially while changing the ENABLE_INACTIVE_SCHEMAS flag and filtering by active/inactive): 
  * EGAP (hidden)
  * Confirmatory - General (hidden)
  * RIDIE Initiation (hidden)
  * RIDIE Study Complete (hidden) 
  * AEA_Preregistration (deleted)
  * Replication Recipe (Brandt et al., 2013): Post-Completion **v1** (deleted)
  * Replication Recipe (Brandt et al., 2013): Pre-Registration **v1** (deleted)
  * Open-Ended Registration **v1** (deleted)
  * OSF-Standard Pre-Data Collection Registration **v1** (deleted)

#### Side Effects
* I believe one of the RIDIE schemas was enabled on staging1 for a period of time -- those registrations may be broken in some way after this.
* @sloria, if any of the staging environments happen to have registrations associated with the v1 schemas (unlikely), the 0147 migration will fail. 

#### Ticket
[PLAT-1246](https://openscience.atlassian.net/browse/PLAT-1246)
